src/main/scala/fs2/async/immutable/Signal.scala

```diff
@@ -0,0 +1,102 @@
+package fs2.async.immutable
+
+import fs2.{Async, process1, Stream}
+import fs2.util.Functor
+
+import fs2.async.{immutable, mutable}
+
+
+/**
+ * Created by pach on 10/10/15.
+ */
+trait Signal[F[_],A]  {
+  /**
+   * Returns a continuous stream, indicating whether the value has changed.
+   * This will spike `true` once for each time the value of `Signal` was changed.
+   * It will always start with `true` when the process is run or when the `Signal` is
+   * set for the first time.
+   */
+  def changed: fs2.Stream[F, Boolean]
+
+  /**
+   * Returns the discrete version stream of this signal, updated only when `value`
+   * is changed.
+   *
+   * Value _may_ change several times between reads, but it is
+   * guaranteed this will always get latest known value after any change.
+   *
+   * If you want to be notified about every single change use `async.queue` for signalling.
+   *
+   * It will emit the current value of the Signal after being run or when the signal
+   * is set for the first time
+   */
+  def discrete: fs2.Stream[F, A]
+
+  /**
+   * Returns the continuous version of this signal, always equal to the
+   * current `A` inside `value`.
+   *
+   * Note that this may not see all changes of `A` as it
+   * gets always current fresh `A` at every request.
+   */
+  def continuous: fs2.Stream[F, A]
+
+  /**
+   * Returns the discrete version of `changed`. Will emit `Unit`
+   * when the `value` is changed.
+   */
+  def changes: fs2.Stream[F, Unit]
+
+  /**
+   * Discrete signal yielding to `true` when this signal halts.
+   * Otherwise emits single `false` and awaits signal to be halted and then emits `true`.
+   * @return
+   */
+  def closed: fs2.Stream[F,Boolean]
```

Comment:
Re stopping several signals at once, not sure what you mean. If signals are always infinite, you can still use them to control termination of several streams, right? Maybe I need to see a more concrete use case.